### PR TITLE
Add toasts to UI and PluginApi

### PR DESCRIPTION
**Pull request by rauenzi**
I also added it to the global bd object for debugging/testing. Also have it popup a toast on setting changed for debugging purposes.

Haven't worked with Vue too much so let me know how bad I fucked up.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-- stable
+- "8"
```
Comment (samuelthomas2774) [inline .travis.yml]: I did try using Gulp v4 with Node.js v10.0.0 and it worked fine. Maybe we should be updating Gulp instead of downgrading Node.js?

Comment (rauenzi) [reply .travis.yml]: Yeah that works for me, v4 should be stable enough

File: client/src/ui/components/BdToasts.vue [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * Copyright (c) 2015-present Jiiks/JsSucks - https://github.com/Jiiks / https://github.com/JsSucks
+ * All rights reserved.
+ * https://betterdiscord.net
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+*/
+
+<template>
+    <bd-toasts>
```
[the remark this answers is not in the document]
Comment (rauenzi) [reply client/src/ui/components/BdToasts.vue]: I made the toasts by looking at the modals and I believe they used a custom element as well. At least that's how I see it registered in the other file and in the DOM as well.

File: client/src/ui/components/common/Toast.vue [hunk 3]
```diff
@@ -0,0 +1,46 @@
+/**
+ * BetterDiscord Modal Component
```
[the remark this answers is not in the document]
Comment (rauenzi) [reply client/src/ui/components/common/Toast.vue]: Oops, lazy copy paste

File: client/src/ui/toasts.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
+/**
+ * BetterDiscord Modals
```
[the remark this answers is not in the document]
Comment (rauenzi) [reply client/src/ui/toasts.js]: Another lazy copy paste

File: client/src/ui/components/BdToasts.vue [hunk 5]
```diff
@@ -0,0 +1,31 @@
+/**
+ * BetterDiscord Modals Component
```
[the remark this answers is not in the document]
Comment (rauenzi) [reply client/src/ui/components/BdToasts.vue]: More lazy copy paste

File: client/src/modules/pluginapi.js [hunk 6]
```diff
@@ -289,6 +289,27 @@ export default class PluginApi {
             }
         });
     }
+	
+	
+	/**
```
[the remark this answers is not in the document]
Comment (rauenzi) [reply client/src/modules/pluginapi.js]: That's what I get for trying to change editor while working lol. I can do toasts as an object

**Comment by rauenzi** [comment]
Approve of toasts in that fork, closing this.

**Pull request closed without merging**